### ISSUE TITLE
Add support for alle worfklow_dispatch input types

### ISSUE THREAD

**Issue by valentingregoire** [rhysd/actionlint#553]
There are `workflow_displatch` input types that are not supported. According to the [docs](https://docs.github.com/en/enterprise-cloud@latest/actions/reference/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype), we should be able to select more than just the provided types provided in the error. You can also find this in the [reference](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype) that is provided in the official [schema](https://www.schemastore.org/github-workflow.json) (see `workflowDispatchInput.type`).

This is the error I receive:

```shell
.github/workflows/some-file.yaml:10:15: invalid value "choice" for input type of workflow_call event. it must be one of "boolean", "number
", or "string" [syntax-check]
   |
10 |         type: choice
   |               ^~~~~~
.github/workflows/some-file.yaml:13:9: unexpected key "options" for "inputs at workflow_call event" section. expected one of "default", "d
escription", "required", "type" [syntax-check]
   |
13 |         options:
   |         ^~~~~~~~
------
```

As you can see, the `options` namespace should also be supported of course.

**Comment by holtkampjs** [comment]
@valentingregoire 

A `workflow_call:` event trigger does not support `type: choice`. `workflow_dispatch:` and `workflow_call:` are similar, but based on the error message you supplied:

> .github/workflows/some-file.yaml:10:15: invalid value "choice" for input type of workflow_call event. ...

I can see this is under a `workflow_call:` trigger which is documented to only support [`boolean`, `number`, or `string` types][workflow-call-input-types].

It is also for that reason, I believe, that `options:` is not supported as a key.

If this were a `workflow_dispatch:` trigger, I believe both issues would be resolved.

[workflow-call-input-types]: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onworkflow_callinputsinput_idtype

**Comment by valentingregoire** [comment]
You are right, thanks for the clarification!
